tests/Unit/Mail/RemindStudentTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Tests\Unit\Mail;
+
+use Tests\TestCase;
+use App\Mail\RemindStudent;
+use Illuminate\Support\Facades\Mail;
+use App\Models\User;
+
+class RemindStudentTest extends TestCase
+{
+    protected $user;
+    protected $remindStudentMail;
+
+    public function setUp() : void
+    {
+        parent::setUp();
+        $this->user = factory(User::class)->make();
+        $this->remindStudentMail = new RemindStudent($this->user);
+    }
+
+    public function tearDown() : void
+    {
+        unset($this->user);
+        unset($this->remindStudentMail);
+        parent::tearDown();
+    }
+
+    public function test_build_return_markdown_with_data()
+    {
+        Mail::fake();
+        Mail::send($this->remindStudentMail);
+        Mail::assertSent(RemindStudent::class);
+        Mail::assertSent(RemindStudent::class, function ($mail) {
+            $mail->build();
+            $this->assertEquals($mail->viewData['username'], $this->user->name);
+            $this->assertEquals($mail->viewData['url'], route('home'));
+
+            return true;
+        });
+    }
```

Comment:
giải thích hàm này

Reply:
Hàm này em dùng để kiểm tra việc gửi mail và kiểm tra dữ liệu truyền vào markdown.
`Mail::fake();`: Đoạn này em đùng để fake Mail (sau này tiến hành gửi và assert)
`Mail::send($this->remindStudentMail);`: Đoạn này em tiến hành gửi Mail
`Mail::assertSent(RemindStudent::class);`: em kiểm tra xem mail được gửi đi chưa
`$mail->build();`: Đoạn này em tiến hành build mail;
`Mail::assertSent(RemindStudent::class, function ($mail) {
            $mail->build();
            $this->assertEquals($mail->viewData['username'], $this->user->name);
            $this->assertEquals($mail->viewData['url'], route('home'));

            return true;
        });`: em kiểm tra xem thư  gửi đi có được gửi có được truyền vào đúng data là tên của người nhận và route('home') hay không?